File: app/src/main/java/com/gmail/laktionov/sample/rx/lyricsfinder/splash/ui/SplashActivity.java
package com.gmail.laktionov.sample.rx.lyricsfinder.splash.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.gmail.laktionov.sample.rx.lyricsfinder.R;
import com.gmail.laktionov.sample.rx.lyricsfinder.search.ui.SearchActivity;

import java.util.concurrent.TimeUnit;

import io.reactivex.Completable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class SplashActivity extends AppCompatActivity {

    private Intent intent;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        delayStart();
    }

    private void delayStart() {
        Completable.fromRunnable(() -> {
            intent = new Intent(this, SearchActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        })
                .subscribeOn(Schedulers.computation())
                .observeOn(AndroidSchedulers.mainThread())
                .delay(2, TimeUnit.SECONDS)
                .subscribe(() -> {
                    startActivity(intent);
                    finish();
                });
    }

}

File: app/src/main/java/com/gmail/laktionov/lyricsfinder/splash/SplashActivity.kt
package com.gmail.laktionov.lyricsfinder.splash

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.animation.ObjectAnimator
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.gmail.laktionov.lyricsfinder.R
import com.gmail.laktionov.lyricsfinder.search.SearchActivity
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        delayStart()
    }

    private fun delayStart() {

        val animator = ObjectAnimator
                .ofFloat(app_label, View.ALPHA, 0f, 1f)
                .setDuration(1_500L)

        animator.addListener(object : AnimatorListenerAdapter() {
            override fun onAnimationEnd(animation: Animator?) {
                val searchIntent = Intent(this@SplashActivity, SearchActivity::class.java).also {
                    it.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                }
                startActivity(searchIntent)
                finish()
            }
        })
        animator.start()
    }
}
